tomplot/data_manipulation.py

```diff
@@ -21,11 +24,13 @@ def area_restriction(field_data, coords_X, coords_Y, coord_lims):
     """
 
     if len(np.shape(field_data)) != 1:
-        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame.')
     if len(np.shape(coords_X)) != 1:
-        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame.')
     if len(np.shape(coords_Y)) != 1:
-        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame.')
+    if len(coord_lims) in (0,2):
```

Comment:
I think this is not actually doing what we need it to. This is checking if the variable `coord_lims` (which should be a dictionary), has a length of 0 or 2. I think we want to be checking the length of the *keys* in the dictionary:
```
if len(coord_lims.keys()) == 0:
    ...
```

Reply:
so from what i understand writing len(dic) is the same as writing (len(dic.keys()).  
More then happy to change it but i think the default when calling len on a dictionary is number of keys